cmd/evm/runner.go

```diff
@@ -206,6 +206,7 @@ func runCmd(ctx *cli.Context) error {
 	execTime := time.Since(tstart)
 
 	if ctx.GlobalBool(DumpFlag.Name) {
+		statedb.Commit(true)
```

Comment:
Are you quite certain that the order here is correct?

Reply:
I don't think the ordering matters, it seems like the root hash result is the same either way.

Reply:
Ok, I don't think there will be any difference unless there are either suicides or empty touch-deletes involved. In those cases, `true` signals to use state-clearing rules, meaning that this will be incorrect for e..g frontier rules. So if you provide a genesis where frontier is disabled, this will not be correct. A more correct approach would be to use
```
  statedb.Commit(chainConfig.IsEIP158(block.Number())
```